### PR TITLE
Fixes in HANA provider

### PR DESCRIPTION
**Pull request by mrylov**
This pull request contains several improvements and bug fixes found in HANA provider.

### CODE REVIEW

File: src/providers/hana/qgshanaconnection.cpp [hunk 1]
```diff
@@ -490,6 +622,9 @@ QVector<QgsHanaLayerProperty> QgsHanaConnection::getLayers(
       layer.srid = -1;
       layer.type = isGeometryColumn ? QgsWkbTypes::Type::Unknown : QgsWkbTypes::NoGeometry;
 
+      if ( layerFilter != nullptr && !layerFilter( layer ) )
```
Comment (nyalldawson) [inline src/providers/hana/qgshanaconnection.cpp:625]:
```suggestion
      if ( layerFilter && !layerFilter( layer ) )
```

File: src/providers/hana/qgshanaconnection.h [hunk 2]
```diff
@@ -28,7 +28,23 @@
...
+  QString schemaName;
+  QString tableName;
+  QString name;
+  short type;
+  QString typeName;
+  int size;
+  int precision;
+  bool isAutoIncrement;
+  bool isNullable;
+  bool isSigned;
```
Comment (nyalldawson) [inline src/providers/hana/qgshanaconnection.h]: The non-string members should be initialized here

**Review by nyalldawson — APPROVED** [review]

**Comment by mrylov** [comment]
Thank you @nyalldawson for the prompt review.
Would that be possible to backport these changes into 3.18?
Is there a way to tell the bot to backport my further changes automatically?

**Comment by nyalldawson** [comment]
In this case you'd need to manually backport and open a new PR -- the backport bot doesn't like rebases

**Pull request merged**